src/ClusterMap.ts

```diff
@@ -1113,23 +1113,20 @@ export default class ClusterMap implements IVisual {
     }
 
     private loadSelectionFromPowerBI() {
-        if (this.lastSelectionArgs !== null) {
+        if (this.lastSelectionArgs && this.lastSelectionArgs[0]) {
             const key = this.lastSelectionArgs[0].key;
-            if (this.lastSelectionArgs) {
-                const personaData = this.dataLayerStack[this.dataLayerStack.length - 1].data.personas.find(p => p.select.key === key);
-                if (personaData) {
-                    const sender = this.personas.personas.find(p => p.id === personaData.id);
-                    if (sender) {
-                        this.handleSelection(sender.object);
-                    }
+            const personaData = this.dataLayerStack[this.dataLayerStack.length - 1].data.personas.find(p => p.select.key === key);
+            if (personaData) {
+                const sender = this.personas.personas.find(p => p.id === personaData.id);
+                if (sender) {
+                    this.handleSelection(sender.object, true);
                 }
             }
         }
     }
 
-    private handleSelection(sender) {
+    private handleSelection(sender, shouldSelect) {
```

Comment:
What's the reasoning behind this change, is it just semantics?

Reply:
I needed to force it to 'true' when coming from either update or bookmarks, since the data will have been freshly loaded without the selection state, but i still need the click handler to use whatever the interaction specified

Reply:
Makes sense, thanks